src/test/util/chainstate.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+//
+#ifndef BITCOIN_TEST_UTIL_CHAINSTATE_H
+#define BITCOIN_TEST_UTIL_CHAINSTATE_H
+
+#include <clientversion.h>
+#include <fs.h>
+#include <node/context.h>
+#include <node/utxo_snapshot.h>
+#include <rpc/blockchain.h>
+#include <validation.h>
+
+#include <univalue.h>
+
+#include <boost/test/unit_test.hpp>
+
+auto NoMalleation = [](CAutoFile& file, SnapshotMetadata& meta){};
```

Comment:
In commit "move-only: unittest: add test/util/chainstate.h" (529f5055675f6c7f734796a8339761b6c1ba13b9)

Can declare this `const auto` instead of `auto` to avoid link errors marco and achow are reporting

- https://github.com/bitcoin/bitcoin/pull/21526#issuecomment-918540412
- https://github.com/bitcoin/bitcoin/pull/21526#issuecomment-919945778
- https://stackoverflow.com/questions/18113164/lambda-in-header-file-error

Reply:
Done! And yet again @ryanofsky bails me out of C++ jail.

Reply:
Indeed this fixed the linker error.